### PR TITLE
Added to_buffer method and updated save method

### PR DESCRIPTION
**Pull request by campaul**

### CODE REVIEW

**Comment by campaul** [comment]
@achennu Once this PR is merged you'll be able to do the following:

```python
# Load a RAW file to a numpy array. Data is represented as RGB triples.
# For example: na[0], na[1], and na[2] represent the red, green, and blue
# values of the first pixel respectively.


from rawkit.raw import Raw
import numpy


with Raw(filename='/some/raw/file.CR2') as raw:
    # Manually get the width and height from the libraw_data_t struct
    width = raw.data.contents.sizes.width
    height = raw.data.contents.sizes.height

    # "Develop" the image to RGB data
    raw.process()

    # Load the image data to a numpy array
    na = numpy.frombuffer(raw.to_buffer(), numpy.int8)
```

**Comment by campaul** [comment]
Closes #4

File: rawkit/raw.py [hunk 1]
```diff
@@ -52,14 +53,30 @@ def process(self, options=None):
...
         """
-        Save the raw data as a new PPM or TIFF image. This method is provided
+        Save the image data as a new PPM or TIFF image. This method is provided
         for convenience, but in general export should actually be handled by
         another library (such as Wand).
 
         Keyword arguments:
-        filename -- A filename (ending in ``.ppm`` or ``.tiff``) to save.
+        filename -- A filename to save.
+        filetype -- A filetype (``ppm`` or ``tiff``).
```
Comment (SamWhited) [inline rawkit/raw.py]: Can you tweak these to use the format like the docstring on the Raw object? It's not as nice looking here, but it makes the docs way easier to parse. Eg.

```
:param filename: the name of a raw file to load                                                                      
:type filename: :class:`basestring`                                                                                  
:returns: A raw object                                                                                               
:rtype: :class:`Raw`
```

**Comment by achennu** [comment]
@campaul, thanks for the heads up. I just tested it and it works great. For completeness of converting the buffer into a RGB channel image, I leave this code snippet here.

from rawkit.raw import Raw
import numpy


with Raw(filename='/some/raw/file.CR2') as raw:
    # Manually get the width and height from the libraw_data_t struct
    width = raw.data.contents.sizes.width
    height = raw.data.contents.sizes.height
    chans = raw.data.contents

    # "Develop" the image to RGB data
    raw.process()

    # Load the image data to a numpy array
    na = numpy.frombuffer(raw.to_buffer(), numpy.int8)
   
    # num of color channels
    chans = na.size/(width*height)   
# can also be read in from metadata

**Comment by achennu** [comment]
Some follow-up issues:
- The raw.process() method converts the sensor data into uint8. How does on read out the full 16-bit data, to possibly get the 12-bit intensity info in there. (something like the dcraw -6 option)
- The RGB channels I read out are totally off-color, and I think more than just because of a missing white balance. Is the order of the channels RGB correct in the first place?

**Comment by campaul** [comment]
> The raw.process() method converts the sensor data into uint8. How does on read out the full 16-bit data, to possibly get the 12-bit intensity info in there. (something like the dcraw -6 option)

I'll work on adding access to the 16-bit data tomorrow, it looks like it shouldn't take too long to knock out.

> The RGB channels I read out are totally off-color, and I think more than just because of a missing white balance. Is the order of the channels RGB correct in the first place?

The data is in RGB order. Can you provide the sample image you're using so I can test it?

**Comment by campaul** [comment]
One thing you might try to fix the color issue is forcing camera white balance while developing.

```python
raw.data.contents.params.use_camera_wb = 1
```

**Pull request merged**